ta4j-core/src/main/java/org/ta4j/core/indicators/bollinger/BollingerBandFacade.java

```diff
@@ -76,9 +77,24 @@ public BollingerBandFacade(Indicator<Num> indicator, int barCount, Number k) {
         this.lower = middle.minus(stdev.multipliedBy(k));
     }
 
+    /**
+     * Create the BollingerBands facade based on the VWAP
+     *
+     * @param vwap     the volume weighted average price indicator
+     * @param barCount the number of periods used for the indicators
+     * @param k        the multiplier used to calculate the upper and lower bands
+     */
+    public BollingerBandFacade(VWAPIndicator vwap, int barCount, Number k) {
```

Comment:
@hhashim1 added a constructor specifically for use with the `VWAPIndicator`. I originally wanted to (re)use the existing `public BollingerBandFacade(Indicator<Num> indicator, int barCount, Number k)` however it always applies a `SMAIndicator`, which - while fine for most "base" indicators - is not wanted for a VWAP. 

@team172011 do you have any opinions here? Refactor the existing constructor (smaller, cleaner code but potentially a breaking change for clients) or the current approach (less elegant but guaranteed safe/backwards compatible). @majgil feel free to add your 2 cents as well.

Reply:
I add my 2 cents. That refactoring deserves to have its own pull-request.